pkg/eestream/decode.go

```diff
@@ -86,12 +86,10 @@ func (dr *decodedReader) Read(p []byte) (n int, err error) {
 		dr.currentStripe++
 	}
 
-	// copy what data we have to the output
+	// copy what data we have to the output, and shrink the remaining buffer
 	n = copy(p, dr.outbuf)
-	// slide the remaining bytes to the beginning
-	copy(dr.outbuf, dr.outbuf[n:])
-	// shrink the remaining buffer
-	dr.outbuf = dr.outbuf[:len(dr.outbuf)-n]
+	dr.outbuf = dr.outbuf[n:len(dr.outbuf)]
```

Comment:
I'm not quite certain that this will be more performant in practice. But, I'm not certain that it's slower either. The previous approach avoids reallocating buffers, and I think `p` should usually read most of the `outbuf`... so it actually won't end up copying a lot.

So, I would like to see a benchmark showing the difference, either in allocations or performance.

Also it can be just `dr.outbuf = dr.outbuf[n:]`.

Reply:
More performant may not, without benchmarking it isn't something known, but mostly the change is, at least for me, because it's less lines of code and simpler to understand.

> Also it can be just dr.outbuf = dr.outbuf[n:].

Yes, that's even better.

Reply:
The code is more complex than it should be and I would guess someone who writes the complex version, probably knows the simpler version. This suggest to me, that there was a reason why the complex version was chosen over the simpler one.

Reply:
it sounds reasonably, my thoughts were that it was something written that it escaped a code review and you know sometimes when focusing on the code and with some back and forth those things sometimes land without that much polishing.